demo/src/components/demos/icon.js

```diff
@@ -0,0 +1,255 @@
+/* global window */
+import React, {Component} from 'react';
+import DeckGL, {IconLayer, WebMercatorViewport} from 'deck.gl';
+import autobind from 'autobind-decorator';
+import rbush from 'rbush';
+
+import {MAPBOX_STYLES} from '../../constants/defaults';
+import {readableInteger} from '../../utils/format-utils';
+
+const ICON_SIZE = 80;
+
+function getIconName(size) {
+  if (size === 0) {
+    return '';
+  }
+  if (size < 10) {
+    return `marker-${size}`;
+  }
+  if (size < 100) {
+    return `marker-${Math.floor(size / 10)}0`;
+  }
+  return 'marker-100';
+}
+
+function stopPropagation(evt) {
+  evt.stopPropagation();
+}
+
+export default class IconDemo extends Component {
+
+  static get data() {
+    return [
+      {
+        url: 'data/film-locations.txt',
+        worker: 'workers/film-locations-decoder.js'
+      },
+      {
+        url: 'images/location-icon-mapping.json'
+      }
+    ];
+  }
+
+  static get parameters() {
+    return {
+      cluster: {displayName: 'Cluster', type: 'checkbox', value: false}
+    };
+  }
+
+  static get viewport() {
+    return {
+      mapStyle: MAPBOX_STYLES.DARK,
+      longitude: -89.4,
+      latitude: 36.7,
+      zoom: 3,
+      maxZoom: 20,
+      pitch: 0,
+      bearing: 0
+    };
+  }
+
+  static renderInfo(meta) {
+    return (
+      <div>
+        <h3>Where Films Were Made (2007 - 2017)</h3>
+        <p>Click a pin to expand detials.</p>
+        <p>Data source: <a href="http://www.imdb.com/conditions">IMDB</a></p>
+        <div className="layout">
+          <div className="stat col-1-2">Movies
+            <b>{ readableInteger(meta.movies || 0) }</b>
+          </div>
+          <div className="stat col-1-2">Locations
+            <b>{ readableInteger(meta.locations || 0) }</b>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  constructor(props) {
+    super(props);
+
+    // build spatial index
+    this._tree = rbush(9, ['.x', '.y', '.x', '.y']);
```

Comment:
The source data come in the format of `[{name, scene, coordinates}]`. The projected positions of the coordinates are calculated in `_updateCluster()` and tagged onto each datum as `x` and `y`.

I use the projected positions instead of longitude and latitude to build this spatial index (because this particular dataset is distributed all over the world, we can't use some fixed deltaLon and deltaLat).

Will add more comments for this.